kf_lib_data_ingest/etl/stage_analyses.py

```diff
@@ -57,28 +47,92 @@ def check_counts(discovery_sources, expected_counts):
                 elif key.UNIQUE_KEY in discovery_sources:
                     key = key.UNIQUE_KEY
 
-            found = uniques.get(key)
+            found = uniques_without_na.get(key)
             if expected == found:
-                passmark = '✅'
+                flag = ''
             else:
                 passed = False
-                passmark = '❌'
+                flag = '❌'
             checks = checks.append(
                 {
-                    'key': key, 'expected': expected, 'found': found,
-                    'equal': passmark
+                    'Key': key, 'Expected': expected, 'Found': found,
+                    'Errors': flag
                 },
                 ignore_index=True
             )
-        message.append(
+        messages.append(
+            'EXPECTED COUNT CHECKS:\n' +
             tabulate(
                 checks,
                 headers='keys', showindex=False, tablefmt='psql'
             )
         )
 
-    return passed, '\n'.join(message)
+    return passed, messages
+
+
+def compare_counts(
+    name_one, discovery_sources_one, name_two, discovery_sources_two
+):
+    title = 'COMPARING COUNTS'
+    messages = [title + '\n' + '='*len(title)]
+    passed = True
+
+    one_keys = discovery_sources_one.keys()
+    two_keys = discovery_sources_two.keys()
+
+    comparison_df, diff_num = _compare(name_one, one_keys, name_two, two_keys)
+
+    if diff_num != 0:
+        msg = f'❌ Column names not equal between {name_one} and {name_two}'
+        messages.extend(_format(msg, comparison_df))
+        passed = False
+
+    for k in (one_keys | two_keys):
+        one_k_keys = discovery_sources_one.get(k, {}).keys()
+        two_k_keys = discovery_sources_two.get(k, {}).keys()
+
+        comparison_df, diff_num = _compare(name_one, one_k_keys,
+                                           name_two, two_k_keys)
+
+        if diff_num != 0:
+            msg = (
+                f'❌ Column values for {k} not equal between {name_one} and '
+                f'{name_two}\n'
+                f'Number of different values = {diff_num}'
+            )
+            messages.extend(_format(msg, comparison_df))
+            passed = False
+
+    return passed, messages
+
+
+def _compare(name_one, list_one, name_two, list_two):
+    indicator = 'Errors'
+    one = pandas.DataFrame({name_one: list(list_one)}).dropna()
```

Comment:
You can filter nulls while constructing the list so that you don't have to iterate over the same values twice. See #276

Reply:
more verbose but ok